modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/management_bar/components/CreationMenu.js

```diff
@@ -25,7 +25,7 @@ function CreationMenu({primaryItems}) {
 	const dataSetContext = useContext(DataSetDisplayContext);
 
 	return (
-		!!primaryItems?.length && (
+		!!primaryItems?.length > 0 && (
```

Comment:
This feels like a weird pattern... do we need the `!!` now?

@wincent, @brunobasto, I think you said you know how to read this 😂

Reply:
wrong merge after rebase! I apologise :(

Reply:
😘

Reply:
> This feels like a weird pattern... do we need the `!!` now?
> 
> @wincent, @brunobasto, I think you said you know how to read this 😂

In general I only do `!!` when I _need_ a boolean (if a mere truthy/falsey value will suffice I use it).

Examples of where you _need_ a boolean:

- Somewhere TypeScript tells you you need a boolean. 😂
- When you want to be sure that React doesn't take something falsey like `0` and render it as `0` instead of nothing.

Examples of where a truthy/falsey will do:

- `if` and ternary conditions.
- `Array.prototype.filter` (and friends) callback return values.

About this specific example, I see @FabioDiegoMastrorilli changed it from:

```js
!!primaryItems?.length &&
```

to:

```js
primaryItems?.length > 0 &&
```

This was that React case I was talking about above. `&&` is a bit treacherous with React. Because if the left hand expression evaluates to `0` (falsey), React will Render `0`. If it evaluates to boolean `false`, it will render nothing. So you have to be careful with `&&`. A ternary is generally harder to misuse. I made https://github.com/liferay/eslint-config-liferay/issues/194 to explore creating a lint for this.

Reply:
If I remember correctly, the `!!` here was recently added by me to avoid this exact problem of rendering zero. Like Fabio said, I think he must have rebased and didn't notice the `> 0` was no longer needed.

Reply:
@FabioDiegoMastrorilli based on @wincent's comment, can you change it back to `!!`? I was about to run ci-forward but if we wanna fix this, better push before running that :)